Refactor: Add comprehensive assertions for system initialization

serialexperimentlain opened the pull request:
This pull request introduces a series of assertions across multiple source files to improve code readability and ensure correct system startup.

The main purpose of these changes is to explicitly document the expected initial states and conditions for various hardware components and variables, making the code more robust and easier to maintain.

**Specific Changes:**

- **CompicMain.c**: Added assertions for pin states (B_6, B_7, RSTPIC_RESTART_PIN, TRXPWR_PIN), transceiver modes, and the Watchdog Timer (WDT) range.

- **FABmain.c**: Added assertions for initial pin states (B_6, B_7) and WDT range.

- **MainPICmain.c**: Added assertions for RTC value ranges and the state of the MBOSS_EN pin.

- **ResetMain.c**: Added assertions to verify the state of various power lines and the RTC value range.

- **STARPIC.c**: Added assertions for the initial states of B_6 and B_7 pins.

These changes are purely for documentation and do not alter the program's logic. They are crucial for verifying correct system configuration and facilitating future debugging.

serialexperimentlain committed:
add: Add assertion for initial pin state

This commit adds an assertion to the `Settings()` function to check the initial state of the `B_7` and `B_6` pins. This ensures that the program starts with the expected hardware configuration.

serialexperimentlain committed:
refactor: Improve code readability and fix assertion logic

This commit refactors the `Settings()` function to enhance clarity and maintainability.
- Corrected a logical error in the `B_7` pin assertion.
- Added a new assertion to verify the state of `RSTPIC_RESTART_PIN`.
- Included clarifying comments to explain transceiver mode settings.
- Standardized line endings to CRLF (`^M`) to ensure consistency across different platforms.

変更の理由:

serialexperimentlain committed:
refactor: Add assertions for hardware states and WDT setup

This commit introduces `assert` statements to the `Settings()` function to document the expected states of hardware pins and the Watchdog Timer (WDT) setup. These assertions enhance code clarity, making it easier to verify that the system's initial configuration is correct and robust. The changes include adding assertions for `RSTPIC_RESTART_PIN`, `TRXPWR_PIN`, and the WDT range.

serialexperimentlain committed:
refactor: Add assertions for RTC and MBOSS_EN pin

This commit refactors the `setting()` function to add new assertions that improve code clarity and maintainability.
- Added an assertion to document the expected range for RTC (Real-Time Clock) values, which helps to validate the input parameters for `Write_OBC_RTC`.
- Included assertions for the `MBOSS_EN` pin to confirm its initial and final states, ensuring the Mission Boss is correctly enabled.
- These changes are purely for documentation and do not alter the program's logic.

serialexperimentlain committed:
refactor: Standardize line endings and reformat comments for TRXPWR_PIN

This commit standardizes the line endings for the `TRXPWR_PIN` section in the `Settings()` function to use CRLF (`^M`). It also refactors the comments to improve code alignment and readability without changing the program's logic.

serialexperimentlain committed:
refactor: Add power line state assertions

This commit refactors the `settings()` function to include comprehensive assertions for the power line control states. It adds assertions to document the initial and final states of `MP_CP_BuckBoost`, `MainPic_Power`, `ComPic_Power`, `_3V3Power_Line1`, `_3V3Power_Line2`, `_5V0Power_Line`, `Unreg1_Line`, and `Unreg2_Line`. These changes improve code readability and ensure the power configurations are correctly documented.

serialexperimentlain committed:
clean: Add RTC value range assertion

This commit adds an assertion comment to the `Set_RTC` function to document the valid range for its year, month, day, hour, minute, and second parameters. This change improves code clarity and maintainability by explicitly stating the expected input values for the function.

serialexperimentlain committed:
refactor: Add assertions for initial pin states in main()

This commit refactors the `main()` function to include assertions for the initial states of `PIN_B6` and `PIN_B7`. These assertions improve code readability and document the expected input values for these pins, ensuring a more robust and self-documenting startup process.

serialexperimentlain committed:
refactor: Add assertions for pin states and WDT range

This commit refactors the `Settings()` function by adding assertion comments for the pin states and the Watchdog Timer (WDT) range. The changes include documenting the expected states for `B_6` and `B_7` and validating the time range for the WDT. This improves code readability and maintainability without altering the program's core logic.

zacky1972 commented on Source_Codes/COM_PIC/CompicMain.c:
```diff
@@ -24,13 +24,23 @@ char B_6, B_7 ;
 
 void Settings()
 {
+   // assert(B_6 = 0 || B_6 = 1);
```
The equality sign in a condition is written as `==` instead of `=`.

```c
// assert(B_6 == 0 || B_6 == 1);
```

The same applies below.

zacky1972 commented on Source_Codes/COM_PIC/CompicMain.c:
```diff
@@ -41,7 +51,8 @@ void Settings()
    ENABLE_DISABLE_NEW_TRX_POWER();                   // check new trx should be on or not
-      
+   
+   // assert(4ms <= WDT <= 4194000ms);
```
Concatenated inequalities use logical conjunction.
We need to look at how milliseconds need to be achieved.

```c
// assert(4 <= WDT && WDT <= 4194000);
```

The same applies below.

serialexperimentlain committed:
refactor: Standardize assert comparisons in Settings function

This commit refactors the `Settings` function to standardize the comparison operator used in assert comments from `=` to `==` for proper logical evaluation. This change improves code clarity and consistency, ensuring that assertions accurately reflect boolean comparisons rather than assignment operations.

Additionally, the WDT range assertion is updated to use the logical AND operator (`&&`) for a more precise and correct representation of the condition.

serialexperimentlain committed:
refactor: Standardize assert comparisons in Settings function

This commit refactors the `Settings` function to standardize the comparison operator used in assert comments from `=` to `==`. This change improves code clarity and consistency, ensuring that assertions accurately reflect boolean comparisons rather than assignment operations.

serialexperimentlain committed:
refactor: Standardize assert comparisons in `setting()`

This commit refactors the `setting()` function to standardize the comparison operator used in assert comments from `=` to `==`. This change improves code clarity and consistency, ensuring that assertions accurately reflect boolean comparisons rather than assignment operations.

Additionally, the RTC date and time assertion is updated to use the logical AND operator (`&&`) for a more precise and correct representation of the condition. This provides a more robust way to document the valid range for the RTC values.

serialexperimentlain committed:
refactor: Standardize assert comparisons in `settings()`

This commit refactors the `settings()` function to standardize the comparison operator used in assert comments from `=` to `==`. This change improves code clarity and consistency, ensuring that assertions accurately reflect boolean comparisons rather than assignment operations.

Additionally, the RTC date and time assertion is updated to use the logical AND operator (`&&`) for a more precise and correct representation of the condition. This provides a more robust way to document the valid range for the RTC values.

serialexperimentlain committed:
refactor: Standardize assert comparisons in main()

This commit refactors the `main()` function to standardize the comparison operator used in assert comments from `=` to `==`. This change improves code clarity and consistency, ensuring that assertions accurately reflect boolean comparisons rather than assignment operations.

serialexperimentlain committed:
refactor: Standardize assert comparisons in `setting()`

This commit refactors the `setting()` function to standardize the comparison operator used in assert comments from `=` to `==`. This change improves code clarity and consistency, ensuring that assertions accurately reflect boolean comparisons rather than assignment operations.

Additionally, the RTC date and time assertion is updated to use the logical AND operator (`&&`) for a more precise and correct representation of the condition. This provides a more robust way to document the valid range for the RTC values.

zacky1972 commented on Source_Codes/Main PIC/MainPICmain.c:
```diff
@@ -27,12 +27,16 @@ Void setting()
    
    SETUP_RTC(RTC_ENABLE | RTC_CLOCK_SOSC,0);   // enabling internal RTC of main pic
+   
+   // assert(0 <= year && year < 100, 1 <= month && month < 13, 1 <= day && day < 32, 0 <= hour && hour < 24, 0 <= minute && minute < 60, 0 <= second && second < 60);
```
Separate assert according to variables:

```c
// assert(0 <= year && year < 100);
// assert(1 <= month && month <= 12);
// assert(1 <= day && day <= 31);
// assert(0 <= hour && hour < 24);
// assert(0 <= minute && minute < 60);
// assert(0 <= second && second < 60);
```

zacky1972 commented on Source_Codes/Reset_PIC/ResetMain.c:
```diff
@@ -53,6 +75,7 @@ void settings()
    SETUP_ADC(ADC_CLOCK_INTERNAL);
    SETUP_ADC_PORTS(sAN2|sAN1|sAN0|sAN4|sAN6|sAN9);   // setting all analog ports
    
+   // assert(0 <= year && year < 100, 1 <= month && month < 13, 1 <= day && day < 32, 0 <= hour && hour < 24, 0 <= minute && minute < 60, 0 <= second && second < 60);
```
Separate assert according to variables:

```c
// assert(0 <= year && year < 100);
// assert(1 <= month && month <= 12);
// assert(1 <= day && day <= 31);
// assert(0 <= hour && hour < 24);
// assert(0 <= minute && minute < 60);
// assert(0 <= second && second < 60);
```

serialexperimentlain committed:
refactor: Standardize assert comparisons in `setting()`

This commit refactors the `setting()` function to separate a combined RTC assertion into individual lines. This improves readability by making each condition explicit.

The change replaces a single multi-conditional assertion with several distinct assertions, one for each component of the date and time (year, month, day, hour, minute, second). This makes the code easier to understand and debug.

serialexperimentlain committed:
refactor: Standardize assert comparisons in `setting()`

This commit refactors the `setting()` function to separate a combined RTC assertion into individual lines. This improves readability by making each condition explicit.

The change replaces a single multi-conditional assertion with several distinct assertions, one for each component of the date and time (year, month, day, hour, minute, second). This makes the code easier to understand and debug.

zacky1972 commented on Source_Codes/FAB/FABmain.c:
```diff
@@ -11,9 +11,13 @@ char B_6, B_7 ;
+   // assert(B_7 == 0 || B_7 == 1);
    B_7 = input(PIN_B7);
    
+   // // assert(4 <= WDT && WDT <= 4194000);
```
Duplicate the symbols `//`.

zacky1972 commented on Source_Codes/Reset_PIC/ResetMain.c:
```diff
@@ -31,16 +31,38 @@ Void TIMER1_ISR()
 void settings()
 {
    //output_low(PIN_C6);
+   // assert(int1 status == 0 || int1 status == 1);
```
What does mean `int1`?
If you uncomment this code and compile it, you will get an error and need to fix it.

serialexperimentlain replied:
In the descriptions of many registers, there are notations like "1 = Bit is set" and "0 = Bit is cleared," which directly correspond to the possible values of the `int1` type. From this, it appears the `int1` data type is a proprietary data type for PIC microcontroller C compilers, used for handling 1-bit boolean values (0 or 1). This allows for the direct and efficient control of hardware on/off states and similar conditions. For example, in the `MP_CP_BuckBoost` function in `ResetMain.c`, an `int1` type argument is used with macro values like `ON` or `OFF`, which helps in directly controlling the on/off state of the buck-boost converter.

zacky1972 replied:
Any type declaration should not be written in assertion, so it should be `assert(status == 0 || status == 1);`.
If you want to declare the type `int1`, write as follows:

```c
int1 status;
assert(status == 0 || status == 1);
```

But it still has an issue that `status` is not defined.

serialexperimentlain committed:
refactor: Fix WDT assertion comment

This commit fixes a typo in the WDT assertion comment within the `Settings()` function. The previous comment had a redundant forward slash (`// //`), which has been corrected. This change ensures the comment is properly formatted and accurately documents the watchdog timer's expected behavior without altering the code's functionality.

serialexperimentlain commented:
The variable `status` is declared as an argument in functions like `MP_CP_BuckBoost`, which are defined within the `ADC_Power_Lines_Control.c` file in the `Reset_PIC` directory.

serialexperimentlain committed:
refactor: Standardize assert variable and operator usage in settings()

This commit refactors the `settings()` function to improve the clarity and consistency of its assert comments. The changes standardize variable names and comparison operators to more accurately reflect the code's logic.

Specifically, this refactor:
- Replaces `int1 status` with the more direct `status` in assert comments.
